ReactSkia/components/RSkComponentTextInput.cpp

```diff
@@ -6,95 +6,175 @@
  */
 
 #include "include/core/SkPaint.h"
+#include "modules/skparagraph/include/TextStyle.h"
+#include "react/renderer/components/textinput/TextInputShadowNode.h"
 #include "ReactSkia/components/RSkComponentTextInput.h"
-#include "ReactSkia/views/common/RSkDrawUtils.h"
+#include "ReactSkia/components/RSkComponent.h"
 #include "ReactSkia/sdk/RNSKeyCodeMapping.h"
-#include "react/renderer/components/textinput/TextInputShadowNode.h"
+#include "ReactSkia/textlayoutmanager/RSkTextLayoutManager.h"
+#include "ReactSkia/views/common/RSkDrawUtils.h"
+#include "ReactSkia/views/common/RSkConversion.h"
 
 #include <string.h>
+#include <iostream>
 
 namespace facebook {
 namespace react {
 
 using namespace RSkDrawUtils;
+using namespace skia::textlayout;
+
+#define NUMBER_OF_LINES         1
+#define FONTSIZE_MULTIPLIER     10
 
 RSkComponentTextInput::RSkComponentTextInput(const ShadowView &shadowView)
-    : RSkComponent(shadowView)
-    ,isInEditingMode_(false){RNS_LOG_INFO("called constructor");}
+    : RSkComponent(shadowView),
+    isInEditingMode_(false) {}
+
+void drawTextInput(SkCanvas *canvas,
+  LayoutMetrics layout,
+  std::shared_ptr<ParagraphBuilder> &builder,
+  const TextInputProps& props) {
+  Rect frame = layout.frame;
+  ParagraphStyle paraStyle;
+  float yOffset;
+
+  // setParagraphStyle
+  paraStyle.setMaxLines(NUMBER_OF_LINES);
+  paraStyle.setEllipsis(u"\u2026");
+  builder->setParagraphStyle(paraStyle);
+
+  // buildParagraph
+  std::shared_ptr<Paragraph> paragraph = builder->Build();
+  paragraph->layout(layout.getContentFrame().size.width);
+
+  // clipRect and backgroundColor
+  canvas->clipRect(SkRect::MakeXYWH(frame.origin.x, frame.origin.y, frame.size.width, frame.size.height));
+  canvas->drawColor(RSkColorFromSharedColor(props.backgroundColor, SK_ColorTRANSPARENT));
+
+  // paintParagraph
+  yOffset = (layout.getContentFrame().size.height - paragraph->getHeight()) / 2;
+  paragraph->paint(canvas, frame.origin.x + layout.contentInsets.left, frame.origin.y + layout.contentInsets.top + yOffset);
+}
 
 void RSkComponentTextInput::OnPaint(SkCanvas *canvas) {
   auto component = getComponentData();
   auto const &textInputProps = *std::static_pointer_cast<TextInputProps const>(component.props);
   /*Retrieve Shadow Props*/
   RNS_LOG_TODO("shadow color,offset,opacity,Radius should be taken from layer and convet into the Skia formate and update here");
-  /* TODO shadow color,offset,opacity,Radius should be taken from layer and convet into the 
-  * Skia formate and update here.
-  ShadowMetrics shadowMetrics{};
```

Comment:
As draw shadow is not handled, suggest to remove this shadowMetric handling, as that is removed is one other PR in review. ON merge both will leads to compilation error